guides/v2.1/ui_comp_guide/concepts/ui_comp_data_source.md

```diff
@@ -0,0 +1,113 @@
+    </item>
+</argument>
+``` 
+
+This will include `Magento/Ui/view/base/web/js/form/provider.js` on the page as part of this DataSource component. The Form Provider Javascript can also be extended if the functionality doesn't do what is necessary in your case. 
+
+Remember that this data provider is still, technically speaking, a completely separate UI Component. There are a few things that need to happen in order to fully link it to the "base" component so that it can use the data provided by the data provider. 
+
+First, we need to declare a "provider" in our base component so it will be able to find that data provider component. Under the `<argument name="data" />` node, add a node like this (where `[ComponentName]` is the name of the component):
+
+```
+<item name="js_config" xsi:type="array">
+    <item name="provider" xsi:type="string">[ComponentName].[ComponentName]_data_source</item>
+</item>
+```
+
+Take note of that item as it is a useful way to connect the UI component, not just the data provider, but other components and sub-components as well.
+
+# Javascript Template Literals
+
+Throughout Magento's core Javascript components there are strings like this: `'${ $.provider }:data.totalRecords'`. These are ES2015 [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals). The `${ }` surrounds an expression that will be parsed as Javascript. `$.provider` is the expression, in this example. 
+
+The template literal is in a single-quote string, however, not the back-ticks which are standard for ES2015. As a result, they would normally be treated as a string. Since Magento needs to support browsers that don't recognize template literals a special interpreter was built for them. If the browser does support standard back-tick template literals, Magento will use that, and if not, it evaluates the value manually. 
+
+When the component is initialized, it will automatically evaluate all string literals in properties. In the example above, `$.provider` will become  `[ComponentName].[ComponentName]_data_source`. The value of this `provider` was declared in the `js_config` block in the XML above. It is possible to pass nearly any configuration value this way and access it using template literals. 
+
+But, XML is static and while that gets us the name of the data provider component, it still doesn't actually provide data. There is one more important step in providing data to Javascript components.
+
+# Javascript Component Linking
+
+Every Javascript component should extend the core Element class in some way (mapped to `uiElement` with RequireJS and located in `Magento/Ui/view/base/web/js/lib/core/element/element.js`.  When this class initializes it runs an `initLinks()` method. That method, in turn, passes a few class properties into a method that handles linking components together. This file (`lib/core/element/link.js`) binds the values of those parameters to actual components. 
```

Comment:
A link to `uiElement` about-page http://devdocs.magento.com/guides/v2.1/ui_comp_guide/concepts/ui_comp_uielement_concept.html